### PR TITLE
feat: add endpoints to call Vipps Login APIs

### PR DESCRIPTION
**Pull request by Nisha1306**
Endpoints to login user through Vipps login API
Ref https://github.com/AtB-AS/kundevendt/issues/1991

### CODE REVIEW

File: src/api/vipps-login/index.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        options: {
+            description: 'Get Vipps user custom token',
+            validate: postVippsLoginRequest
+        },
+        handler:  async (request, ) => {
+            const client = await getClient();
+            const {authorizationCode, state, nonce} = (request.payload as unknown) as VippsCustomTokenRequest;
+            const params = await client.callback(VIPPS_CALLBACK_URL,
+                { code: authorizationCode, state: state},
+                { nonce: nonce, state: state}
```
Comment (torfjor) [inline src/api/vipps-login/index.ts:53]: Are you validating the stored state in the callback handler in app? If so, I guess it's okay to pass the same variable twice :grin:

File: src/api/vipps-login/index.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            )
+            if(params.access_token) {
+                const userInfo = await client.userinfo(params)
+                const phoneNumber = userInfo.phone_number;
+                if(phoneNumber){
+                    const auth = app.auth();
+                    try{
+                        const user = await auth.getUserByPhoneNumber(`+${phoneNumber}`)
+                        return await auth.createCustomToken(user.uid);
```
Comment (torfjor) [inline src/api/vipps-login/index.ts:62]: Some spacing issues? Needs a format?

**Review by torfjor — APPROVED** [review]

File: src/api/vipps-login/index.ts (hunk 1 above)
Comment (Nisha1306) [reply src/api/vipps-login/index.ts:53]: validating in the app 👍

**Comment by torfjor** [comment]
I pushed changes to the deployment. Just merge whenever you're ready 🤞

**Pull request merged**